### DISCUSSION TITLE
EXPAND module arguments

### DISCUSSION THREAD

**Discussion by Hannibal-Vecteur** [Yubico/pam-u2f#313] [Q&A] [answered]
Hello,

I am trying to use the module argument EXPAND into /etc/pam.d/sudo file and this is what I try to do:

`auth sufficient pam_u2f.so expand authfile=/etc/.u2f_keys/$USER/u2f_keys cue nouserok debug`

and this is what I get when i do sudo echo test :

`Home directory for yolo is /home/yolo
Using authentication file /etc/.u2f_keys/$USER/u2f_keys
debug(pam_u2f): util.c:925 (get_devices_from_authfile): Cannot open file: /etc/.u2f_keys/$USER/u2f_keys (Aucun fichier ou dossier de ce type)
`

As specified into the README, I tried %u and %%USER

Any idea please ?

**Comment by LDVG** [comment, marked as answer]
Hi,

`%u` would be the correct variable; i.e. 

```
authfile=/etc/.u2f_keys/%u/u2f_keys
```

This feature is available starting from pam-u2f 1.3.0. If you're using an earlier version, you'll need to update first.
